### DISCUSSION TITLE
How to resolve "INFO: No distance passed to HantushWellModel, assuming r=1.0."

### DISCUSSION THREAD

**Discussion by pgraafstra** [pastas/pastas#437] [Q&A] [answered]
Hey!
Finally got around to trying the clustered well approach as per https://www.frontiersin.org/articles/10.3389/feart.2022.907609/full
Only... it seems like passing distances=[], to ps.stressmodels.WellModel(ps.HantushWellModel) while not resulting in an error, does result in "INFO: No distance passed to HantushWellModel, assuming r=1.0.", and I've been unable to resolve this.
Would love some pointers!

**Comment by raoulcollenteur** [comment]
Hi ! Good to hear you are trying out the new WellModel for multiple wells.

 Looks like you did not provide distance from the monitoring well to the pumping wells (r), and therefore Pastas assumes the distance are 1 meter. So I think the quick 'fix' is to provide distance, and you're set! 

`ps.stressmodels.WellModel(stress=discharge, rfunc=ps.HantushWellModel, name="Wells, distance=[1000, 500])`

Hope this helps.

Cheers,
Raoul

**Reply by pgraafstra** [reply]
Hey, thanks for the quick response!
I have done exactly that, as far as I'm aware. Simplified example of my line to add the WellModel:
```
ml.add_stressmodel(
   ps.stressmodels.WellModel(stress=[pdSeries1, pdSeries2],
   rfunc=ps.rfunc.HantushWellModel, 
   distances = [6000, 15000],
   name='extractions_WVP1'))
```

**Comment by dbrakenhoff** [comment, marked as answer]
Hi,

Nice that you're using WellModel! Can you tell where in your code the message is being generated? 

As the message (vaguely) suggests this is not a problem, but something the user should potentially be aware of. HantushWellModel accepts 4 parameters (`A`, `a`, `b`, and the distance `r`). If the distance `r` is not passed, it will still work but then it assumes `r=1`. This happens for example when you plot the results (`ml.plots.results()`) because it will plot the step response, but because there are multiple wells, we've chosen to plot the step response at `r=1`. That causes this message to be shown, but is not something to be worried about. 

When you want to calculate the gain at a particular well for example, it is important you also pass the distance to `HantushWellModel.gain()` so you get the gain at the well (or distance) you intended. If you forget the distance you will see this message, which then hopefully reminds you that you should be including the distance. So that's why it was implemented.

Hope that helps?

Davíd

P.S. Raouls message just popped up I see, but posting this anyway in case it helps clarify even more.

**Reply by pgraafstra** [reply]
Thanks for the additional context... As I understand from your explanation, distances should only influence gain outcomes?
Based on Raoul's article and my own assumptions on how the functions are implemented, I would expected the distance to be a vital parameter in fitting timeseries. I would expect not providing distances to result in all extractions to be considered of equal influence when modelling the timeseries, while providing distances would result in a scaling of the influences over distance.

Thus: if a 1000m3 extraction at 6k distance would cause a 15 cm effect on my modelled groundwaterlevels, a 1000m3 extraction at 15k distance would result in a 6 cm effect.
When not providing distances, I would expect the model to be unable to make this distinction, thus creating them all as equal by setting the same assumed distance for all wells, and resulting in a less accurate representation of the hydrological system.

**Reply by dbrakenhoff** [reply]
The distance affects the shape of the step response, not just the gain. That was an example of when it is important for the user to know the distance should be passed to `HantushWellModel.gain()`.

And you're right in your interpretation, but it shouldn't be possible to use the WellModel without any distances...? If you want your wells to have equal impact, you'd need to pass identical distances.

Also, I noticed the INFO message is also generated by the fit_report after running `ml.solve()`, once again nothing to be worried about, but I can understand if that is a bit confusing. I'll see if i can figure out a way to silence the message for situations where internal pastas calls are triggering it.

**Reply by pgraafstra** [reply]
Hey Davíd,
Thanks for your response. It triggered me to do some additional testing. It seems to me like it is functioning as it should, and the INFO message popping up is indeed caused by the fit_report, while actually using the distances provided, causing different fits when I change the distances provided, but not the extraction rates.
